components/concept/fetch/src/main/java/mozilla/components/concept/fetch/Response.kt

```diff
@@ -44,10 +44,28 @@ data class Response(
      * A [Body] returned along with the [Request].
      *
      * **The response body can be consumed only once.**.
+     *
+     * @param stream the input stream from which the response body can be read.
+     * @param contentType optional content-type as provided in the response
+     * header. If specified, an attempt will be made to look up the charset
+     * which will be used for decoding the body. If not specified, or if the
+     * charset can't be found, UTF-8 will be used for decoding.
      */
     open class Body(
-        private val stream: InputStream
+        private val stream: InputStream,
+        contentType: String? = null
     ) : Closeable, AutoCloseable {
+
+        @Suppress("TooGenericExceptionCaught")
+        private val charset = contentType?.let {
+            val charset = it.substringAfter("charset=")
```

Comment:
Ah, that's nice and simple. I wonder whether there are situations where this could not be the last value? Something like:

`Content-Type: text/html; charset=utf-8; foobar`

But I can't find any evidence that this is the case. :)

Reply:
Yeah, I was looking too and couldn't find a case where this should break!